### PR TITLE
cody: fix view load condition

### PR DESCRIPTION
**Pull request by abeatrix**
Close https://github.com/sourcegraph/sourcegraph/issues/54045

Notice we were setting authStatus on config change, which would first pick up the default config when the view is first loaded.

Since we now actually have a Login logic that log users in and out on config change that is related to auth (endpoint, token, customheader), where config is shared with the webview on every text doc change to update codebase context (we should look into refactoring this too 😓 ), we should not update auth status on the config change logic (which is actually text doc change unrelated to auth), and rely on the `login` logic for auth status and use that to update view instead.

Here you can see what is called when we first load the extension:
image

## Test plan

Here is a video of how the changes would look:

https://github.com/sourcegraph/sourcegraph/assets/68532117/3875d28c-1ee9-4ab5-9334-cd0154462d44

### CODE REVIEW

File: client/cody/webviews/App.tsx [hunk 1]
```diff
@@ -54,25 +55,20 @@ export const App: React.FunctionComponent<{ vscodeAPI: VSCodeWrapper }> = ({ vsc
                 }
                 case 'config':
                     setConfig(message.config)
-                    setAuthStatus(message.authStatus)
```
Comment (abeatrix) [inline client/cody/webviews/App.tsx:57]: don't change view and auth status in config load

Comment (philipp-spiess) [reply client/cody/webviews/App.tsx:57]: Should we remove it from the `massage` payload then?

File: client/cody/webviews/App.tsx [hunk 2]
```diff
@@ -42,6 +42,7 @@ export const App: React.FunctionComponent<{ vscodeAPI: VSCodeWrapper }> = ({ vsc
         vscodeAPI.onMessage(message => {
             switch (message.type) {
                 case 'transcript': {
+                    console.log('transcript', message)
```
Comment (philipp-spiess) [inline client/cody/webviews/App.tsx:45]:
```suggestion
```

File: client/cody/webviews/App.tsx [hunk 3]
```diff
@@ -54,25 +55,20 @@ export const App: React.FunctionComponent<{ vscodeAPI: VSCodeWrapper }> = ({ vsc
...
                     setView(isLoggedIn(message.authStatus) ? 'chat' : 'login')
-                    if (message.authStatus.endpoint) {
-                        setEndpoint(message.authStatus.endpoint)
-                    }
+                    setEndpoint(message.authStatus.endpoint)
                     break
                 case 'showTab':
                     if (message.tab === 'chat') {
-                        setView('chat')
+                        setView(message.tab)
```
Comment (philipp-spiess) [inline client/cody/webviews/App.tsx:67]: This change is redundant, just checking you made it intentionally

**Review by philipp-spiess — APPROVED** [review]

**Comment by abeatrix** [comment]
moved to https://github.com/sourcegraph/sourcegraph/pull/54106 cause i dont know how to revert changes 🤡

**Pull request closed without merging**